conda_lock/pypi_solver.py

```diff
@@ -333,12 +341,34 @@ def _prepare_repositories_pool(allow_pypi_requests: bool) -> Pool:
     """
     factory = Factory()
     config = factory.create_config()
-    repos = [
-        factory.create_legacy_repository(
-            {"name": source[0], "url": source[1]["url"]}, config
-        )
-        for source in config.get("repositories", {}).items()
-    ]
+
+    # read auth from global poetry config
+    auth_config_file = TOMLFile(Path(user_config_dir("pypoetry")) / "auth.toml")
+    if auth_config_file.exists():
+        config.merge(auth_config_file.read())
```

Comment:
the auth.toml gets created by `poetry config http-basic.<private-source-name> <user> <password>`, see: https://python-poetry.org/docs/repositories/#installing-from-private-package-sources. The code here assumes that for each source you added to your your pyproject.toml that needs login, you provided those by `poetry config` prior to attempting to lock.

Reply:
Oh interesting for my use case i need to be on the companies VPN and i will be able to access to private repo.

Reply:
I am not an expert, so I might be completely wrong here - but I think it shouldn't matter too much whether you are using VPN or not, as long as during locking and environment creation you are in the same VPN. And in case you still need to authenticate, you can do that with http basic authentication and the auth credentials are provided in the auth.toml file.